Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,74 @@ namespace AZ
+            for (const auto& mesh : lodAsset->GetMeshes())
+            {
+                // Get the index buffer data, confirming that the asset is valid and indices are 32 bit integers. Other formats are
+                // currently not supported.
+                const AZ::RPI::BufferAssetView& indexBufferView = mesh.GetIndexBufferAssetView();
+                const AZ::RHI::BufferViewDescriptor& indexBufferViewDesc = indexBufferView.GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> indexBufferAsset = indexBufferView.GetBufferAsset();
+                if (!indexBufferAsset.IsReady() || indexBufferViewDesc.m_elementSize != sizeof(uint32_t))
+                {
+                    continue;
+                }
+
+                // Get the position buffer data, if it exists with the expected name.
+                const AZ::RPI::BufferAssetView* positionBufferView = mesh.GetSemanticBufferAssetView(positionName);
+                if (!positionBufferView)
+                {
+                    continue;
+                }
+
+                // Confirm that the position buffer is valid and contains 3 32 bit floats for each position. Other formats are currently not
+                // supported.
+                const AZ::RHI::BufferViewDescriptor& positionBufferViewDesc = positionBufferView->GetBufferViewDescriptor();
+                const AZ::Data::Asset<AZ::RPI::BufferAsset> positionBufferAsset = positionBufferView->GetBufferAsset();
+                if (!positionBufferAsset.IsReady() || positionBufferViewDesc.m_elementSize != sizeof(float) * 3)
+                {
+                    continue;
+                }
+
+                const AZStd::span<const uint8_t> indexRawBuffer = indexBufferAsset->GetBuffer();
+                const uint32_t* indexPtr = reinterpret_cast<const uint32_t*>(
+                    indexRawBuffer.data() + (indexBufferViewDesc.m_elementOffset * indexBufferViewDesc.m_elementSize));
+
+                const AZStd::span<const uint8_t> positionRawBuffer = positionBufferAsset->GetBuffer();
+                const float* positionPtr = reinterpret_cast<const float*>(
+                    positionRawBuffer.data() + (positionBufferViewDesc.m_elementOffset * positionBufferViewDesc.m_elementSize));
+
+                // Copy the index and position data into the visible geometry structure.
+                AzFramework::VisibleGeometry visibleGeometry;
+                visibleGeometry.m_transform = AZ::Matrix4x4::CreateFromTransform(m_transformInterface->GetWorldTM());
+
+                // This resize and copy assumes the stride between elements is 0.
+                visibleGeometry.m_indices.resize(indexBufferViewDesc.m_elementCount);
+                memcpy(&visibleGeometry.m_indices[0], indexPtr, indexBufferViewDesc.m_elementCount * indexBufferViewDesc.m_elementSize);
+
+                visibleGeometry.m_vertices.resize(positionBufferViewDesc.m_elementCount * 3);
```

Comment:
Using `resize_no_construct` is more efficient as the integer and floating point variables don't have to be initialized to zero in that case